pkg/goDB/DBWorkManager.go

```diff
@@ -65,26 +77,111 @@ type DBWorkManager struct {
 	numProcessingUnits int
 
 	tFirstCovered, tLastCovered int64
+	nWorkloads                  uint64
+
+	keepAlive      time.Duration
+	stats          *WorkloadStats
+	statsCallbacks StatsCallbacks
+}
+
+// StatsCallback is a function which can be used to communicate running workload statistics
+type StatsCallback func(stats *WorkloadStats) error
+
+// StatsCallbaacks is a list of StatsCallbacks
+type StatsCallbacks []StatsCallback
+
+// Execute runs all callbacks
+func (scs StatsCallbacks) Execute(stats *WorkloadStats) error {
+	for _, cb := range scs {
+		err := cb(stats)
+		if err != nil {
+			return err
```

Comment:
This will break on first error, skipping all other functions. If this is what you intend to do, all good.